week4/assignment/week4_assignment/src/pages/MyPage/FixInfo.tsx

```diff
@@ -0,0 +1,98 @@
+import { useEffect, useState } from 'react';
+import { checkingInstance } from '../../api/axios';
+import { useNavigate } from 'react-router-dom';
+import styled from '@emotion/styled';
+import { Theme } from '../../styles/Themes';
+
+const FixInfo = () => {
+  const [password, setPassword] = useState('');
+  const [hobby, setHobby] = useState('');
+  const [myInfo, setMyInfo] = useState({});
+  const navigate = useNavigate();
+
+  const clickFixButton = async () => {
+    if (!password && !hobby) {
+      alert('비밀번호 혹은 취미를 입력해주세요');
+      return;
+    }
+    try {
+      await checkingInstance.put('/user', myInfo);
+      alert('정보 수정에 성공했습니다.');
+      navigate('/');
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  useEffect(() => {
+    if (password && hobby) {
+      setMyInfo({
+        hobby,
+        password,
+      });
+    } else if (password && !hobby) {
+      setMyInfo({ password });
+    } else {
+      setMyInfo({ hobby });
+    }
+  }, [password, hobby]);
```

Comment:
```suggestion
  const clickFixButton = async () => {
    if (!password && !hobby) {
      alert('비밀번호 혹은 취미를 입력해주세요');
      return;
    }
    try {
      await checkingInstance.put('/user', {
        hobby,
        password,
      });
      alert('정보 수정에 성공했습니다.');
      navigate('/');
    } catch (error) {
      console.log(error);
    }
  };

```

useEffect까지 사용하며 myInfo라는 state를 별도로 관리할 필요없이 api 요청을 보낼때 위와같이 보내는 게 더 좋을 것 같습니다! (물론 hobby나 password가 빈칸이라면 객체에서 빼는 처리는 해줘야합니다.)

이렇게 하는게 더 좋은 이유는

1. 불필요한 useEffect 사용
2. 같은 값을 여러 state에서 관리하면서 생길 수 있는 데이터 동기화 문제
    여기서 hobby와 password라는 각 state에 데이터가 있는데, 이를 myInfo라는 별도의 state에도 저장하게 되면 똑같은 hobby와 password가 여러개의 state에 저장되게 됩니다.
    이렇게 되면 데이터가 제대로 동기화 되지 않을 수도 있고, 데이터의 흐름을 파악하기 어려워져 유지보수나 가독성 측면에서도 좋지 않습니다!